src/AutoVizuA11y.tsx

```diff
@@ -96,141 +106,111 @@ const AutoVizuA11y = ({
 	const [arrayConverted, setArrayConverted] = useState<number[]>([]);
 	const [number, setNumber] = useState<number>(1);
 	const [descs, setDescs] = useState<string[]>([]);
+	const [descriptionContent, setDescriptionContent] = useState<string>("Generating description...");
+	const [elements, setElements] = useState<HTMLElement[]>([]);
+
+	const chartRef = useRef<HTMLDivElement>(null);
+	const shortcutGuideRef = useRef<HTMLDivElement>(null);
+
+	let componentId = useAutoId();
+
+	let alertDivRef = useRef<HTMLDivElement>(null);
+	let alertDiv: React.ReactNode = useMemo(
+		() => (
+			<div
+				ref={alertDivRef}
+				className="a11y_alert visually-hidden"
+				role="alert"
+				aria-live="assertive"
+			>
+				{"\u00A0"}
+			</div>
+		),
+		[],
+	);
 
-	if (autoDescriptions) {
-		apiKey = autoDescriptions.apiKey;
-		model = autoDescriptions.model;
-		temperature = autoDescriptions.temperature;
-	}
-
-	if (insights == undefined || !(insights in data[0])) {
-		insights = "";
-	}
-
-	// Generate a unique identifier for this component
-	const componentId = newId();
-
-	const storedLongerKey = `oldLonger_${componentId}`;
-	const storedSmallerKey = `oldSmaller_${componentId}`;
-
-	const handleFocus = () => {
-		//css
-		ref.current!.classList.add("focused");
-		toolTutorial = localStorage.getItem("toolTutorial");
-		if (toolTutorial === "true") {
-			alertDiv = ref.current!.getElementsByClassName("a11y_alert")[0] as HTMLElement | null;
-			if (alertDiv) {
-				alertDiv.textContent =
-					"You just entered an AutoVizually chart." +
-					" For information on how to interact with it, press the question mark key" +
-					" to open the shortcut guide";
+	let chartDescription: React.ReactNode = useMemo(
+		() => (
+			<p
+				style={{ textIndent: "-10000px" }}
+				className="a11y_desc visually-hidden"
+				data-testid="a11y_desc"
+				onFocus={() => {
+					handleFirstFocus({ alertDiv, chartRef, alertDivRef });
+				}}
+				onBlur={(_) => handleBlur(chartRef)}
+				aria-label={descriptionContent}
+			>
+				{descriptionContent}
+			</p>
+		),
+		[descriptionContent],
+	);
+
+	let chart = useMemo(
+		() => React.Children.map(children, (child) => <div>{child}</div>),
+		[children],
+	);
+
+	useEffect(() => {
+		if (chartRef.current) {
+			let newElements: HTMLElement[] = [];
+			if (selectorType.element !== undefined) {
+				newElements = Array.from(chartRef.current.querySelectorAll(selectorType.element));
+			} else {
+				newElements = Array.from(
+					chartRef.current.getElementsByClassName(selectorType.className || ""),
+				) as HTMLElement[];
 			}
-			setTimeout(function () {
-				if (alertDiv) {
-					alertDiv.textContent = "\u00A0";
-				}
-			}, 1000);
-			localStorage.setItem("toolTutorial", "false");
-			toolTutorial = "false";
+			setElements(newElements);
 		}
-	};
+	}, [chartRef, selectorType]);
 
-	const handleBlur = () => {
-		ref.current!.classList.remove("focused");
-	};
+	useEffect(() => {
```

Comment:
I would move this to the top of the component as much as possible because it is a "mount" useEffect, so it runs only once when the component mounts, right?